persist-cli/src/main/java/io/ballerina/persist/nodegenerator/SyntaxTreeGenerator.java

```diff
@@ -108,7 +110,52 @@ public static SyntaxTree createPesistToml(String name) {
         return SyntaxTree.from(textDocument);
     }
 
-    public static HashMap<String, String> readToml(Path configPath, String name) throws BalException {
+    public static HashMap<String, String> populateConfiguration(HashMap<String, String> persistConfig, Path configPath)
+            throws BalException {
+        Path fileNamePath = configPath.getFileName();
+        try {
+            TextDocument configDocument = TextDocuments.from(Files.readString(configPath));
+            SyntaxTree syntaxTree = SyntaxTree.from(configDocument, fileNamePath.toString());
+            DocumentNode rootNote = syntaxTree.rootNode();
+            NodeList<DocumentMemberDeclarationNode> nodeList = rootNote.members();
+            for (String configKey : persistConfig.keySet()) {
+                if (Pattern.matches(PLACEHOLDER_PATTERN, persistConfig.get(configKey))) {
+                    boolean configExists = false;
+                    String[] placeHolderValues = persistConfig.get(configKey).replaceAll("\"", "").split("\\.");
+                    String relatedKey =  placeHolderValues[placeHolderValues.length - 1].replaceAll("}", "");
+                    String placeHolderTable = persistConfig.get(configKey)
+                            .replaceAll("\\$\\{", "").replaceAll("}", "").replaceAll("\\." + relatedKey, "");
+                    for (Object member : nodeList) {
+                        if (member instanceof TableNode) {
+                            TableNode node = (TableNode) member;
+                            if (node.identifier().toSourceCode().trim().equals(placeHolderTable.
+                                    replaceAll("\"", ""))) {
+                                NodeList<KeyValueNode> subNodeList = node.fields();
+                                for (KeyValueNode subMember : subNodeList) {
+                                    if (subMember.identifier().toSourceCode().trim().equals(relatedKey)) {
+                                        persistConfig.put(configKey,
+                                                subMember.value().toSourceCode().trim());
+                                        configExists = true;
+                                    }
```

Comment:
When reading Config.toml configs, process it so as Map<String, DocumentMemberDeclarationNode>. That way there is no need to iterate for each placeholder

Reply:
Changed in PRAM https://github.com/ballerina-platform/persist-tools/pull/65